src/Microsoft.Azure.Batch.SoftwareEntitlement.Client.Native/SoftwareEntitlementClient.cpp

```diff
@@ -742,12 +771,32 @@ std::unique_ptr<Entitlement> GetEntitlement(
         url += '/';
     }
 
-    Curl curl;
-    curl.Post(url + "softwareEntitlements?api-version=2017-05-01.5.0", entitlement_token, requested_entitlement);
-
-    curl.VerifyIntermediateCertificate(url);
+    for (unsigned int retry = 1; retry <= retries; ++retry)
+    {
+        try
+        {
+            return Curl::GetEntitlement(url, entitlement_token, requested_entitlement);
+        }
+        catch (const Curl::CurlException& e)
+        {
+            if (e.GetCode() == CURLE_OPERATION_TIMEDOUT)
+            {
+                Sleep(1000UL * retry);
```

Comment:
Units are milliseconds?

Reply:
Yeah, so 1 second times the retry attempt. Seems reasonable.

Reply:
If you want to be more standard and less Windowsy, you can:

```cpp
std::this_thread::sleep_for(std::chrono::seconds(1));
```

Reply:
Oh, good point on xplat.